internal/pkg/views/notifications.go

```diff
@@ -0,0 +1,175 @@
+package views
+
+import (
+	"context"
+	"fmt"
+	"github.com/lawrencegripper/azbrowse/internal/pkg/style"
+	"github.com/lawrencegripper/azbrowse/pkg/armclient"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/jroimartin/gocui"
+	"github.com/lawrencegripper/azbrowse/internal/pkg/eventing"
+)
+
+func init() {
+	pendingDeletes = []pendingDelete{}
+}
+
+type pendingDelete struct {
+	display string
+	url     string
+}
+
+var pendingDeletes []pendingDelete
+var deleteMutex sync.Mutex // ensure delete occurs only once
+var gui *gocui.Gui
+
+// AddPendingDelete queues deletes for
+// delete once confirmed
+func AddPendingDelete(display, url string) {
+	if url == "" {
+		eventing.SendStatusEvent(eventing.StatusEvent{
+			Failure: true,
+			Message: "Item `" + display + "` doesn't support delete",
+			Timeout: time.Second * 5,
+		})
+		return
+	}
+	gui.Update(func(g *gocui.Gui) error {
+		deleteMutex.Lock()
+		defer deleteMutex.Unlock()
+
+		for _, i := range pendingDeletes {
+			if i.url == url {
+				eventing.SendStatusEvent(eventing.StatusEvent{
+					Failure: true,
+					Message: "Item already `" + display + "` in pending delete list",
+					Timeout: time.Second * 5,
+				})
+				return nil
+			}
+		}
+
+		pendingDeletes = append(pendingDeletes, pendingDelete{
+			url:     url,
+			display: display,
+		})
+
+		return nil
+	})
+}
+
+// ConfirmDelete delete all queued/pending deletes
+func ConfirmDelete() {
+	gui.Update(func(g *gocui.Gui) error {
+		deleteMutex.Lock()
+		defer deleteMutex.Unlock()
+
+		pending := pendingDeletes
+		event, _ := eventing.SendStatusEvent(eventing.StatusEvent{
+			InProgress: true,
+			Message:    "Starting to delete items",
+			Timeout:    time.Second * 15,
+		})
+
+		for _, i := range pending {
```

Comment:
Obviously not performance optimal but probably gives the most intuiative user experience i.e. sequential deletes and notifications for failures.

Reply:
Yeah I also kinda think it's not the type of action I'm going to be super worried about going instantly, if anything the slightly slower nature lets you bail out with a `CTRL+C` if you realise you messed up the list.